lib/mqtt/mqtt_client.dart

```diff
@@ -0,0 +1,141 @@
+import "package:flutter/foundation.dart";
+import "package:flutter_dotenv/flutter_dotenv.dart";
+import "package:mqtt_client/mqtt_client.dart";
+import "package:mqtt_client/mqtt_server_client.dart";
+import "package:oss_surveys_customer/mqtt/model/status_message.dart";
+import "package:typed_data/typed_buffers.dart";
+import "../main.dart";
+
+/// MQTT Client
+class MqttClient {
+  
+  late final MqttServerClient _client;
+  final statusTopic = "oss/status";
+  
+  /// Public constructor.
+  /// 
+  /// Initializes [MqttServerClient]
+  MqttClient() {
+    var mqttBasePath = dotenv.env["MQTT_URL"];
+    var mqttClientId = dotenv.env["MQTT_PORT"];
+    var mqttPort = int.tryParse(dotenv.env["MQTT_PORT"] ?? "");
+    _client = MqttServerClient.withPort(mqttBasePath!, mqttClientId!, mqttPort!);
+  }
+  
+  /// Connects MQTT Client if not already connected.
+  Future<void> connect() async {
+    if (_client.connectionStatus!.state == MqttConnectionState.connected) {
+      logger.info("MQTT Client already connected");
+      return;
+    }
+    
+    _client.logging(on: true);
+    _client.onConnected = onConnected;
+    _client.onDisconnected = onDisconnected;
+    _client.onUnsubscribed = onUnsubscribed;
+    _client.onSubscribed = onSubscribed;
+    _client.onSubscribeFail = onSubscribeFail;
+
+    final connMessage = MqttConnectMessage()
+        .keepAliveFor(60)
+        .withWillTopic(statusTopic)
+        .withWillMessage(StatusMessage(false).toJson().toString())
+        .startClean()
+        .withWillQos(MqttQos.atLeastOnce);
+    _client.connectionMessage = connMessage;
+    try {
+      await _client.connect();
+    } catch (e) {
+      logger.shout("Exception: $e");
+      _client.disconnect();
+    }
+
+    _client.updates?.listen((List<MqttReceivedMessage<MqttMessage>> c) {
+      final MqttPublishMessage message = c[0].payload as MqttPublishMessage;
```

Comment:
Maybe you could first check if c[0] is present and not null?

Reply:
Resolved by checking if c is not empty.